### PR TITLE
ref: Rename SentryClient and SentryUser

### LINKED ISSUE #1100: Change SentryUser Swift name to something other than User
The Obj-C `SentryUser` has a Swift name of `User` via `NS_SWIFT_NAME(User)`. `User` is an incredibly common class name in apps, and having this renamed to `User` in Swift causes a lot more issues than it needs to. If I have a class that uses both Sentry and my own `User` class, I need to explicitly use `MyModule.User` in order to avoid naming conflicts.

I understand this would be a breaking change, but using the Sentry framework should provide as few roadblocks to using our own code as possible. Thanks!

### LINKED ISSUE #1842: Rename Client to SentryClient on Swift
### Description

SentryClient is named Client on Swift, it should be SentryClient to align with SentryHub.

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Renamed `Client` to `SentryClient` and `User` to `SentryUser` for Swift.

## :bulb: Motivation and Context

close #1842 
close #1100 

## :pencil: Checklist

- [x] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] Review from the native team if needed
- [ ] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ This version introduces a dependency on Swift, which only impacts you if you don
 ### Features
 
 - Properly demangle Swift class name (#2162)
+- Renaming Client and User to SentryClient and SentryUser (#2403)
```
Comment (kevinrenskers) [inline CHANGELOG.md:10]: That's not really a feature though. I'm in the same boat with #2404 😅

**Comment by brustolin** [comment]
>Can you please do https://github.com/getsentry/sentry-cocoa/issues/1842 and https://github.com/getsentry/sentry-cocoa/issues/1100 in two different PRs, so we have a cleaner Changelog?

@philipphofmann Can we have two changelogs entries pointing to the same PR?

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,10 @@
 
 This version introduces a dependency on Swift, which only impacts you if you don't use Swift. If your project uses Swift already, this change will not affect you.
 
+### Breaking Changes
+
+- Renaming Client and User to SentryClient and SentryUser (#2403)
```
Comment (brustolin) [inline CHANGELOG.md:9]:
```suggestion
- Renaming Client to SentryClient for Swift (#2403)
- Renaming User to SentryUser for Swift (#2403)
```

**Comment by kevinrenskers** [comment]
I think you need to merge the 8.0.0 branch into this branch, so that the diff makes sense again.

**Review by kevinrenskers — APPROVED** [review]

**Comment by kevinrenskers** [comment]
We're getting a whole bunch of SwiftLint violations in TrendingMovies, all of a sudden.

**Pull request merged**